# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../vdr-image/ffmpeg-version.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -ur image-0.2.7/liboutput/Makefile image-0.2.7.work/liboutput/Makefile
--- image-0.2.7/liboutput/Makefile	2006-01-29 17:04:05.000000000 +0100
+++ image-0.2.7.work/liboutput/Makefile	2007-12-30 12:02:56.000000000 +0100
@@ -17,7 +17,7 @@
 
 INCLUDES += -I$(VDRDIR)/include -I.
 ifdef FFMDIR
-INCLUDES += -I$(FFMDIR)/libavcodec -I$(FFMDIR)/libavutil
+INCLUDES += -I$(FFMDIR)
 endif
 
 DEFINES  += -D_GNU_SOURCE
diff -u image-0.2.7/Makefile image-0.2.7.work/Makefile
--- image-0.2.7/Makefile	2006-08-27 15:51:11.000000000 +0200
+++ image-0.2.7.work/Makefile	2007-12-30 11:57:54.000000000 +0100
@@ -72,8 +72,9 @@
 ### The version number of ffmpeg (taken from "libavcodec/avcodec.h" ):
 
 ifdef FFMDIR
-LIBAVCODECVERSION = $(shell grep "\#define LIBAVCODEC_VERSION " $(FFMDIR)/libavcodec/avcodec.h | \
+LIBAVCODECVERSION = $(shell grep "\#define LIBAVCODEC_VERSION " $(FFMDIR)/avcodec.h | \
                     awk '{ print $$3 }' | cut -d "." -f 1 )
+LIBAVNEWAPI = $(shell echo "$(LIBAVCODECVERSION) >= 51" | bc)
 endif
 
 ### Includes and Defines (add further entries here):
@@ -86,10 +87,8 @@
 INCLUDES += -I$(VDRDIR)/include -I.
 
 ifdef FFMDIR
-INCLUDES += -I$(FFMDIR)/libavcodec -I$(FFMDIR)/libavutil
-LIBS += -L$(FFMDIR)/libavcodec
-ifeq ($(LIBAVCODECVERSION),51)
-LIBS += -L$(FFMDIR)/libavformat -L$(FFMDIR)/libavutil
+INCLUDES += -I$(FFMDIR)
+ifeq ($(LIBAVNEWAPI), 1)
 LIBS += -lavformat -lavutil
 endif
 endif
